server/server.js

```diff
@@ -22,8 +24,71 @@ app.use(express.static(reactStaticDir));
 app.use(express.static(uploadsStaticDir));
 app.use(express.json());
 
-app.get('/api/hello', (req, res) => {
-  res.json({ message: 'Hello World!' });
+app.get('/api/Users', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "Users"
+        order by "UserID"
+    `;
+    const result = await db.query(sql);
+    res.json(result.rows);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/Users/sign-up', async (req, res, next) => {
+  try {
+    const { userName, passWord } = req.body;
+    const hashedPassWord = await argon2.hash(passWord);
+    const params = [userName, hashedPassWord, 'user', 0, 0];
+    const sql = `
+    insert into "Users" ("Username", "Password", "Role", "Latitude", "Longitude")
+    values($1, $2, $3, $4, $5)
+    returning "UserID","Username";
+    `;
+    const results = await db.query(sql, params);
+    res.json(results.rows[0]);
+  } catch (err) {
+    if (err.code === '23505') {
+      console.log('hello');
+      res.json({ message: 'User is Taken' });
+    } else {
+      next(err);
+    }
+  }
+});
+
+app.post('/api/Users/sign-in', async (req, res, next) => {
+  try {
+    const { userName, passWord } = req.body;
+    const sql = `
+    select "UserID",
+    "Password"
+    from "Users"
+    where "Username"=$1 `;
+    const params = [userName];
+    const result = await db.query(sql, params);
+    const users = result.rows[0];
+    if (users !== undefined) {
+      const isMatching = await argon2.verify(users.Password, passWord);
+      if (isMatching === true) {
+        const payload = {
+          userID: users.UserID,
+          userName
+        };
+        const token = jwt.sign(payload, process.env.TOKEN_SECRET);
+        res.status(200).json({ payload, token, message: 'Login successful' });
+      } else {
+        res.json({ message: 'Invalid Login' });
+      }
+    } else {
+      res.json({ message: 'Invalid Login' });
+    }
```

Comment:
All these nested ifs are ugly. Since your catch clause sends exactly the same message, why don't you just detect the error and throw? e.g.,

`if (!users) throw new ClientError(400);`
and so on